073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,364 @@
+  This is because currently the state is inferred by observing various Kubernetes annotations and Java class fields.
+
+These benefits will also make it easier to evolve the management of CA certificates in future (see [Future changes](#future-changes)).
+
+Non-goals of the proposal:
+* This proposal does not change the management of Clients CA.
+  Users can provide custom listener certificates, so allowing integration with a certificate management solution for the Clients CA doesn't bring as many benefits as it does for the Cluster CA.
+* This proposal does not introduce integration with a certificate management solution.
+  The aim of this proposal is to make it easier to allow this kind of integration in future, but assumes that the Cluster CA is still managed by Strimzi and (other than during renewal) there is a single Cluster CA certificate for the cluster.
+
+### New trust-state Kubernetes Secret
+
+A new Kubernetes Secret is introduced which is updated by the cluster operator in its role of managing the cluster (specifically which issued certificates are stored where and rolling pods to use/trust those certificates) to reflect whether the individual CA certificates are trusted and/or in use in the cluster.
+
+```yaml
+kind: Secret
+metdata:
+  name: <CLUSTER_NAME>-cluster-ca-trust-state # e.g. foo-cluster-ca-trust-state
+data:
+  ${fingerprint}: ${state}
+```
+
+The `fingerprint` is the fingerprint/thumbprint of the certificate. I.e. the SHA-1 hash of the ASN.1 DER-encoded form of the X509 certificate, which is a commonly used way of identifying certificates using common tooling.
+The `state` can be any of the states of the CA certificate trust state machine described below.
+
+### CA certificate state machine
+
+The "CA certificate trust state machine" has the following states:
+
+```
+  UNTRUSTED // not yet trusted everywhere it needs to be 
+   |
+   v
+  TRUSTED_UNUSED // trusted everywhere it needs to be, but no EE certificates yet issued with this CA certificate
+   |
+   v
+  TRUSTED_IN_USE // trusted everywhere, EE certificates issued using this CA certificate
+   |
+   v
+  PHASE_OUT // trusted everywhere, but new end-entity certificates will not be issued using this CA certificate
+   |
+   v
+  ZOMBIE // still trusted by some component, but with no issued end-entity certificates in current use.
+```
+
+### New trusted-certs Kubernetes Secret
+
+A new Kubernetes Secret is introduced which is updated by the cluster operator in its role as the receiver of issued certificates to include the CA certificates that the components should currently trust.
+This Kubernetes Secret will be mounted into Kubernetes Pods to use a TrustStore.
+
+```yaml
+kind: Secret
+metdata:
+  name: <CLUSTER_NAME>-cluster-ca-trusted-certs # e.g. foo-cluster-ca-trusted-certs
+type: strimzi.io/trust
```

Comment:
This type field was copied from the original proposal to differentiate between the different Secrets that were used for tracking state of the end-entity certificates and root certificates, however I wonder if given this is only a partial implementation of that proposal we should remove it. @tombentley what do you think? We can always introduce this at a later date if needed.

Reply:
I'm happy if we remove this.